### PR TITLE
Change redis.exists to redis.exists? to avoid problems with redis 4.3

### PR DESCRIPTION
**Pull request by cesarizu**
Currently there's a deprecation warning with redis 4.2:

```
`Redis#exists(key)` will return an Integer in redis-rb 4.3, if you want to keep the old behavior, use `exists?` instead. To opt-in to the new behavior now you can set Redis.exists_returns_integer = true. (/home/travis/build/company/project/vendor/bundle/ruby/2.6.0/gems/split-3.4.1/lib/split/experiment.rb:104:in `new_record?')
```

This PR removes that warning by using the new method name.

### CODE REVIEW

File: lib/split/experiment_catalog.rb [hunk 1]
```diff
@@ -13,7 +13,7 @@ def self.all_active_first
     end
 
     def self.find(name)
-      return unless Split.redis.exists(name)
+      return unless Split.redis.exists?(name)
```
[the remark this answers is not in the document]
Comment (cesarizu) [reply lib/split/experiment_catalog.rb:16]: I pushed a new version that defines `exists?` as an alias for `exists` if the method is not defined. That should provide compatibility with `redis < 4.2`. That alias can be removed if/when split decides to remove `redis <= 4.1` support.

Does this approach seem ok?

[the remark this answers is not in the document]
Comment (andrehjr) [reply lib/split/experiment_catalog.rb:16]: After giving some thought, I decided to increase the minimum required redis-rb to >= 4.2. Sidekiq did the same.

I merged your commit without the monkey patch on https://github.com/splitrb/split/commit/12225656da44bb748c2ed9c67f63846dc5f2593d

Thanks!

**Comment by pioz** [comment]
I still have this warning using `split` 3.4.1 and `redis-rb` 4.2.5. Any update?

**Comment by andrehjr** [comment]
I just released with 4.0.0.pre with that fix. @pioz Can you try that?

**Comment by pioz** [comment]
Yes, it works.

**Pull request closed without merging**